export/metric_to_pb.go

```diff
@@ -0,0 +1,162 @@
+package export
+
+import (
+	a1 "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
+	v1 "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	r1 "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"go.opencensus.io/metric/metricdata"
+)
+
+func metricsToServiceRequest(ms []*metricdata.Metric) *a1.ExportMetricsServiceRequest {
+	metrics := []*v1.Metric{}
+
+	for _, m := range ms {
+		toAppend := metricToProto(m)
+		metrics = append(metrics, toAppend)
+	}
+
+	return &a1.ExportMetricsServiceRequest{
+		Metrics: metrics,
+	}
+}
+
+func metricToProto(m *metricdata.Metric) *v1.Metric {
+	return &v1.Metric{
+		MetricDescriptor: metricToDescriptor(m),
+		Timeseries:       metricToTimeSeries(m),
+		Resource:         metricToResource(m),
+	}
+}
+
+func metricToLabelKeys(m *metricdata.Metric) []*v1.LabelKey {
+	labelKeys := []*v1.LabelKey{}
+
+	for _, lk := range m.Descriptor.LabelKeys {
+		toAppend := &v1.LabelKey{
+			Key:         lk.Key,
+			Description: lk.Description,
+		}
+
+		labelKeys = append(labelKeys, toAppend)
+	}
+	return labelKeys
+}
+
+func metricToDescriptor(m *metricdata.Metric) *v1.MetricDescriptor {
+	return &v1.MetricDescriptor{
+		Name:        m.Descriptor.Name,
+		Description: m.Descriptor.Description,
+		Unit:        string(m.Descriptor.Unit),
+		Type:        v1.MetricDescriptor_Type(m.Descriptor.Type + 1),
```

Comment:
suggest we put this logic in a function that is named in such a way it's obvious why it's doing a +1...